src/panels/lovelace/cards/hui-entity-card.ts

```diff
@@ -0,0 +1,270 @@
+        >
+      `;
+    }
+
+    const showUnit =
+      (!this._config.attribute &&
+        stateObj.state !== UNKNOWN &&
+        stateObj.state !== UNAVAILABLE) ||
+      (this._config.attribute && stateObj.attributes[this._config.attribute]);
+
+    return html`
+      <ha-card
+        @action=${this._handleClick}
+        .actionHandler=${actionHandler()}
+        tabindex="0"
+      >
+        <div class="flex header">
+          <div class="name">
+            <span>${this._config.name || computeStateName(stateObj)}</span>
+          </div>
+          <div class="icon">
+            <ha-icon
+              .icon=${this._config.icon || stateIcon(stateObj)}
+            ></ha-icon>
+          </div>
+        </div>
+        <div class="flex info">
+          <span id="value"
+            >${this._config.attribute
+              ? stateObj.attributes[this._config.attribute] ||
+                this.hass.localize("state.default.unknown")
+              : stateObj.state}</span
+          >${showUnit
+            ? html`
+                <span id="measurement"
+                  >${this._config.unit ||
+                    stateObj.attributes.unit_of_measurement}</span
+                >
+              `
+            : ""}
+        </div>
+        ${this._config.footer
+          ? this.renderHeaderFooter(this._config.footer, "footer")
+          : ""}
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProps.get("_config") as
+      | EntityCardConfig
+      | undefined;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config!.theme);
+    }
+  }
+
+  private renderHeaderFooter(
+    conf: LovelaceHeaderFooterConfig,
+    className: string
+  ): TemplateResult {
+    const element = createHeaderFooterElement(conf);
+    if (this.hass) {
+      element.hass = this.hass;
+    }
+    return html`
+      <div class=${"header-footer " + className}>${element}</div>
+    `;
+  }
+
+  private _handleClick(): void {
+    fireEvent(this, "hass-more-info", { entityId: this._config!.entity });
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        display: flex;
+        flex-direction: column;
+      }
+
+      ha-card {
+        display: flex;
+        flex-direction: column;
+        flex: 1;
+        position: relative;
+        cursor: pointer;
+        overflow: hidden;
+      }
+
+      ha-card:focus {
+        outline: none;
+        background: var(--divider-color);
+      }
+
+      .flex {
```

Comment:
Not a fan of extracting this. It makes reading the CSS a bit more complicated as it now has a dependency on the right classes being set in the HTML.